app/components/pipeline/jobs/table/dataReloader.js

```diff
@@ -101,21 +111,44 @@ export default class DataReloader {
     this.numBuilds = numBuilds;
   }
 
-  start() {
-    this.intervalId = setInterval(() => {
-      if (Object.keys(this.jobCallbacks).length === 0) {
-        return;
-      }
+  start(eventId) {
+    if (eventId) {
+      this.workflowDataReload.registerBuildsCallback(
+        QUEUE_NAME,
+        eventId,
+        builds => {
+          this.parseEventBuilds(builds);
+        }
+      );
+    } else {
+      this.intervalId = setInterval(() => {
+        if (Object.keys(this.jobCallbacks).length === 0) {
+          return;
+        }
 
-      this.fetchBuildsForJobs(this.jobIdsMatchingFilter).then(() => {});
-    }, ENV.APP.BUILD_RELOAD_TIMER);
+        this.fetchBuildsForJobs(this.jobIdsMatchingFilter).then(() => {});
+      }, ENV.APP.BUILD_RELOAD_TIMER);
+    }
   }
 
-  destroy() {
+  stop(eventId) {
     if (this.intervalId) {
       clearInterval(this.intervalId);
     }
+    if (eventId) {
+      this.workflowDataReload.removeBuildsCallback(QUEUE_NAME, eventId);
+    }
 
     this.intervalId = null;
   }
+
+  parseEventBuilds(eventBuilds) {
+    eventBuilds.forEach(eventBuild => {
+      const { jobId } = eventBuild;
+      const builds = [eventBuild];
+
```

Comment:
do we need to ensure this?
```suggestion
    // Ensure this.builds is defined
    if (!this.builds) {
      this.builds = {};
    }
```

Reply:
No need, `this.builds` is initialized already during class construction